scripts/keyframes.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+"""Given a gfxreconstruct command file, identify "interesting"
+keyframes based on distinct maximal function sets used in a frame.  Keyframes
+selected in this way tend to exercise more distinct parts of the trace,
+as many frames will use the same "footprint" of functions as other frames;
+by collapsing to maximal function sets, keyframes can be selected that more
+thoroughly exercise the graphics API.
+
+The selected keyframes can be further refined through the use of filter
+operations.
+
+To use on a trace captured by gfxreconstruct, the trace must be converted
+to an ASCII command file containing one function or method name per line, e.g.:
+
+    ...
+    vkDestroyImageView
+    vkBeginCommandBuffer
+    vkCmdPipelineBarrier
+    vkCmdWriteTimestamp
+    vkAcquireNextImageKHR
+    vkCmdPipelineBarrier
+    vkCmdBeginRenderPass
+    vkCmdBindPipeline
+    vkCmdBindDescriptorSets
+    vkCmdSetViewport
+    vkCmdBindIndexBuffer
+    vkCmdDrawIndexed
+    vkCmdPipelineBarrier
+    vkCmdWriteTimestamp
+    vkCmdEndRenderPass
+    vkCmdResetQueryPool
+    vkEndCommandBuffer
+    vkFlushMappedMemoryRanges
+    vkResetFences
+    vkQueueSubmit
+    vkQueuePresentKHR
+    ...
+
+The command file must include frame marker functions (e.g. vkQueuePresentKHR
```

Comment:
I note we have lost the information on where images are being presented (a gui 3d modelling app may present to lots of surfaces in odd orders and we only care about the main viewport presents for example). Perhaps we should use annotations or a metacommand to emit frame markers that are converted to json and then preserved in the text file input to this script. Not a priority: this way is fine for typical games.

Reply:
I note that `gfxrecon-replay` has a `--surface-index` switch.  Perhaps `gfxrecon-convert` should also have a similar switch, or perhaps the surface index should be part of the `gfxrecon-convert` output JSON.